app/src/main/java/cm/aptoide/pt/home/bundles/ads/AdMapper.java

```diff
@@ -1,24 +1,21 @@
 package cm.aptoide.pt.home.bundles.ads;
 
+import android.util.Pair;
 import cm.aptoide.pt.ads.data.AptoideNativeAd;
 import cm.aptoide.pt.search.model.SearchAdResult;
-import java.util.AbstractMap;
-import rx.functions.Func1;
 
 public class AdMapper {
   public AdMapper() {
   }
 
-  public Func1<? super AdClick, AbstractMap.SimpleEntry<String, SearchAdResult>> mapAdToSearchAd() {
-    return wrappedAd -> {
-      if (wrappedAd == null) {
-        return new AbstractMap.SimpleEntry<>("", new SearchAdResult());
-      }
-      AptoideNativeAd ad = (AptoideNativeAd) wrappedAd.getAd();
-      return new AbstractMap.SimpleEntry<>(wrappedAd.getTag(),
-          new SearchAdResult(ad.getAdId(), ad.getIconUrl(), ad.getDownloads(), ad.getStars(),
-              ad.getModified(), ad.getPackageName(), ad.getCpcUrl(), ad.getCpdUrl(), ad.getCpiUrl(),
-              ad.getClickUrl(), ad.getAdTitle(), ad.getAppId(), ad.getNetworkId(), false));
-    };
+  public Pair<String, SearchAdResult> mapAdToSearchAd(AdClick wrappedAd) {
```

Comment:
Maybe create an object to wrap this. It's harder to read this code having a pair here since one of the elements is a string, making the purpose of it unclear